crates/vhost-user-backend/src/handler.rs

```diff
@@ -367,6 +367,10 @@ where
     }
 
     fn set_vring_base(&mut self, index: u32, base: u32) -> VhostUserResult<()> {
+        if index as usize >= self.num_queues {
```

Comment:
My first attempt would have been: `let [mut] vring = self.vrings.get(index).ok_or_else(...)?`?

Removes the double indexing and does not rely on some variable being in sync with the vector.

Reply:
I honestly did direct copy and paste from the other methods that all do this.

That said, I like your approach much better, but I would send another patch in another PR where we change them all together. What do you think?

Reply:
I did it here: https://github.com/rust-vmm/vhost/pull/214

I don't know whether to merge this patch with that at this point and just have the other PR...

Reply:
I am confused, the other PR now lists this as dependency, but this is not updated? :thinking: Got no opinion on doing this through one or two PRs :)

Reply:
@Ablu sorry ;-)
I mean, the other PR is based on this (it includes the patch in this PR). Now my doubt is whether to have 2 patches as now (the first one is then overwritten by the second one, but it could be used for some backport), or to have directly only the second patch (the one added in the other PR).

Reply:
Ah! Either sounds good to me.

Reply:
Okay, let's leave them separate for now ;-)